src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -568,8 +569,9 @@ private class Emit(
             xmv := m,
             xvv := xmv.mux(defaultValue(tarray.elementType), v),
             codeB.setup,
-            xmaccum := codeB.m,
-            xvaccum := xmaccum.mux(defaultValue(typ), codeB.v))
+            xmbody := codeB.m,
+            xvaccum := xmbody.mux(defaultValue(typ), codeB.v),
+            xmaccum := xmbody)
```

Comment:
so is the point that a missing next element does not mean the accumulated value is missing, because for example the body could have `isNA` and evaluate to non-missing?

Reply:
more or less, yeah. And the evaluation of `codeB.v` (on line 573) is assuming `xmaccum` to hold the current missingness of the accumulator, so we can't update that variable to the next value before that's calculated.